src/main/java/org/commonjava/storage/pathmapped/metrics/MeasuredPathDB.java

```diff
@@ -0,0 +1,167 @@
+
+    private final MetricRegistry metricRegistry;
+
+    private final String metricPrefix;
+
+    public MeasuredPathDB( PathDB decorated, MetricRegistry metricRegistry, String metricPrefix )
+    {
+        this.decorated = decorated;
+        this.metricRegistry = metricRegistry;
+        this.metricPrefix = metricPrefix;
+    }
+
+    @Override
+    public List<PathMap> list( String fileSystem, String path )
+    {
+        return measure( () -> decorated.list( fileSystem, path ), "list" );
+    }
+
+    @Override
+    public long getFileLength( String fileSystem, String path )
+    {
+        return decorated.getFileLength( fileSystem, path );
+    }
+
+    @Override
+    public long getFileLastModified( String fileSystem, String path )
+    {
+        return decorated.getFileLastModified( fileSystem, path );
+    }
+
+    @Override
+    public boolean exists( String fileSystem, String path )
+    {
+        return measure( () -> decorated.exists( fileSystem, path ), "exists" );
+    }
+
+    @Override
+    public void insert( String fileSystem, String path, Date date, String fileId, long size, String fileStorage,
+                        String checksum )
+    {
+        measure( () -> decorated.insert( fileSystem, path, date, fileId, size, fileStorage, checksum ), "insert" );
+    }
+
+    @Override
+    public boolean isDirectory( String fileSystem, String path )
+    {
+        return measure( () -> decorated.isDirectory( fileSystem, path ), "isDirectory" );
+    }
+
+    @Override
+    public boolean isFile( String fileSystem, String path )
+    {
+        return measure( () -> decorated.isFile( fileSystem, path ), "isFile" );
+    }
+
+    @Override
+    public boolean delete( String fileSystem, String path )
+    {
+        return measure( () -> decorated.delete( fileSystem, path ), "delete" );
+    }
+
+    @Override
+    public String getStorageFile( String fileSystem, String path )
+    {
+        return decorated.getStorageFile( fileSystem, path );
+    }
+
+    @Override
+    public boolean copy( String fromFileSystem, String fromPath, String toFileSystem, String toPath )
+    {
+        return decorated.copy( fromFileSystem, fromPath, toFileSystem, toPath );
+    }
+
+    @Override
+    public void makeDirs( String fileSystem, String path )
+    {
+        measure( () -> decorated.makeDirs( fileSystem, path ), "makeDirs" );
+    }
+
+    @Override
+    public List<Reclaim> listOrphanedFiles( int limit )
+    {
+        return decorated.listOrphanedFiles( limit );
+    }
+
+    @Override
+    public void removeFromReclaim( Reclaim reclaim )
+    {
+        decorated.removeFromReclaim( reclaim );
+    }
+
+    private static final String TIMER = "timer";
+
+    private static final String METER = "meter";
+
+    private void measure( Runnable runnable, String metricName )
+    {
+        if ( metricRegistry != null )
+        {
+            metricRegistry.meter( name( metricPrefix, metricName, METER ) ).mark();
```

Comment:
A timer should already involve a meter, so I'm thinking if this meter is still needed here.